scripts/3.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,513 @@
+
+# ## Load Data
+
+# In[9]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[10]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[11]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'samples.tsv')\nclinical = pd.read_table(path, index_col=0)")
+
+
+# In[12]:
+
+clinical.tail(5)
+
+
+# In[13]:
+
+# Subset the Y matrix to only the genes to be classified
+y_full = Y[classifier_genes_df['entrez_gene_id']]
+
+
+# In[14]:
+
+y_full.columns = classifier_genes_df['gene_symbol']
+y_full = y_full.assign(disease = clinical['disease'])
+
+# This matrix now stores the final y matrix for the classifier (y['indicator'])
+y = y_full.assign(indicator = y_full.max(axis=1))
+
+
+# In[15]:
+
+unique_pos = y.groupby('disease').apply(lambda x: x['indicator'].sum())
+heatmap_df = y_full.groupby('disease').sum().assign(TOTAL = unique_pos)
+heatmap_df = heatmap_df.divide(y_full.disease.value_counts(sort=False).sort_index(), axis=0)
+
+
+# In[16]:
+
+# What is the percentage of different mutations across different cancer types?
+sns.heatmap(heatmap_df);
+
+
+# Visualizing the input data here is key. The heterogeneity of the mutations across tissues is apparent for this particular pathway. In comparison with `TP53` mutations, it appears that Hippo signaling impacts different tissues with higher diversity.
+# 
+# Looking closer at the plots above, it is evident that several tissues do not demonstrate aberrations (at least at the mutation level) in Hippo signaling. Specifically, it appears that cancers with gender specificity like testicular cancer and and prostate cancer are _not_ impacted. Therefore, because of this artificial imbalance, if Cognoma were to include these cancers in the classifier, it **will** key in on gender specific signal (i.e. genes that are only on the Y chromosome, or X inactivation genes).  
+
+# In[17]:
+
+# How many samples in each tissue that have Hippo signaling aberrations
+ind = ['Negatives', 'Positives', 'Positive Prop']
+percent = heatmap_df['TOTAL']
+neg = y.disease.value_counts() - unique_pos
+tissue_summary_df = pd.DataFrame([neg, unique_pos, percent], index=ind, 
+                                 dtype='object').T.sort_values('Positive Prop', ascending=False)
+tissue_summary_df
+
+
+# ## Filter Data by Tissue
+# 
+# This is a crucial step that is different from previous classifiers
+
+# In[18]:
+
+# Technically, these are hyper parameters, but for simplicity, set here
+filter_prop = 0.10
+filter_count = 15
+
+tissue_prop_decision = tissue_summary_df['Positive Prop'] >= filter_prop
+tissue_count_decision = tissue_summary_df['Positives'] >= filter_count
+
+tissue_decision = tissue_prop_decision & tissue_count_decision
+tissue_decision
+
+
+# This filters out:
```

Comment:
This list seems redundant with the output and update-problematic (hardcoded). Would it make sense to just print out this list of filtered tissues?